sdk/iOS/src/MSTable.h

```diff
@@ -129,31 +129,31 @@ extern NSString *const MSSystemColumnDeleted;
 
 /// Sends a request to the Microsoft Azure Mobile Service to return the item
 /// with the given id from the table.
--(void)readWithId:(id)itemId completion:(MSItemBlock)completion;
+-(void)readWithId:(nonnull id)itemId completion:(nullable MSItemBlock)completion;
 
 /// Sends a request to the Microsoft Azure Mobile Service to return the item
 /// with the given id from the table. Addtional user-defined parameters are
 /// sent in the request query string.
--(void)readWithId:(id)itemId
-       parameters:(NSDictionary *)parameters
-       completion:(MSItemBlock)completion;
+-(void)readWithId:(nonnull id)itemId
+       parameters:(nullable NSDictionary *)parameters
+       completion:(nullable MSItemBlock)completion;
 
 /// Sends a request to the Microsoft Azure Mobile Service to return all items
 /// fromm the table that meet the conditions of the given query.
 /// You can also use a URI in place of queryString to fetch results from a URI e.g.
 /// result.nextLink gives you URI to next page of results for a query that you can pass here.
--(void)readWithQueryString:(NSString *)queryString
-                completion:(MSReadQueryBlock)completion;
+-(void)readWithQueryString:(nullable NSString *)queryString
```

Comment:
we allow it, but seems that if you dont want a QS, you'd just use teh readWithCompletion version, so this could be flagged nonnull?